dev/create-release/release-build.sh

```diff
@@ -414,15 +437,15 @@ if [[ "$1" == "publish-release" ]]; then
       -DskipTests $PUBLISH_PROFILES $SCALA_2_10_PROFILES clean install
   fi
 
-  #./dev/change-scala-version.sh 2.12
-  #$MVN -DzincPort=$ZINC_PORT -Dmaven.repo.local=$tmp_repo \
-  #  -DskipTests $SCALA_2_12_PROFILES §$PUBLISH_PROFILES clean install
+  if ! is_dry_run && [[ $PUBLISH_SCALA_2_12 = 1 ]]; then
+    ./dev/change-scala-version.sh 2.12
+    $MVN -DzincPort=$((ZINC_PORT + 2)) -Dmaven.repo.local=$tmp_repo -Dscala-2.12 \
+      -DskipTests $PUBLISH_PROFILES $SCALA_2_12_PROFILES clean install
+  fi
```

Comment:
I still think we want a cleanup step that always changes the scala version back to the 'default' of 2.11 right now.

Reply:
We tried to do it before but it's commented out: https://github.com/apache/spark/pull/22441/files#diff-01ca42240614718522afde4d4885b40dL424

I didn't look into it and seems it's OK as the release script is able to do scala 2.10 build.

Reply:
I believe I added it, commented out, for the time when we enabled a 2.12 build. My concern is just that it leaves the checkout in a state that builds for 2.12. If you tried a 2.11 build afterwards it would fail. But I suppose it would clearly fail. I suppose it will work here as it's the last thing this tries to do, so the end state doesn't matter. Tidy, but optional to set it back.